subsys/bluetooth/gatt_pool.c

```diff
@@ -364,27 +366,23 @@ int bt_gatt_pool_desc_alloc(struct bt_gatt_pool *gp,
 		return -ENOSPC;
 	}
 
-	attr = &gp->svc.attrs[gp->svc.attr_count];
-	memset(attr, 0, sizeof(*attr));
-
-	memcpy(attr, descriptor, sizeof(*attr));
-	attr->uuid = NULL;
-	ret = uuid_register((struct bt_uuid **) &attr->uuid, descriptor->uuid);
+	ret = uuid_register(&uuid, descriptor->uuid);
```

Comment:
What are the pros of not using attr->uuid directly here?

Reply:
Because I moved it up, so attr is not a valid pointer, and attr->uuid would be overwritten anyway. Also I would have to set attr->uuid = NULL before calling.

Reply:
ok. Got it.